pkg/operator/saptunechangesolution_v1.go

```diff
@@ -0,0 +1,160 @@
+//   On success, the current state of the applied solution is collected as the "after" diff.
+//
+// - ROLLBACK:
+//   TBD
+
+type SaptuneChangeSolution struct {
+	baseOperator
+	saptune         saptune.Saptune
+	parsedArguments *saptuneSolutionArguments
+}
+
+func NewSaptuneChangeSolution(
+	arguments OperatorArguments,
+	operationID string,
+	options OperatorOptions[SaptuneChangeSolution],
+) *Executor {
+	saptuneChange := &SaptuneChangeSolution{
+		baseOperator: newBaseOperator(operationID, arguments, options.BaseOperatorOptions...),
+	}
+
+	saptuneChange.saptune = saptune.NewSaptuneClient(
+		support.CliExecutor{},
+		saptuneChange.logger,
+	)
+
+	for _, opt := range options.OperatorOptions {
+		opt(saptuneChange)
+	}
+
+	return &Executor{
+		phaser:      saptuneChange,
+		operationID: operationID,
+	}
+}
+
+func (sc *SaptuneChangeSolution) plan(ctx context.Context) error {
+	opArguments, err := parseSaptuneSolutionArguments(sc.arguments)
+	if err != nil {
+		return err
+	}
+	sc.parsedArguments = opArguments
+
+	if err = sc.saptune.CheckVersionSupport(ctx); err != nil {
+		return err
+	}
+
+	initiallyAppliedSolution, err := sc.saptune.GetAppliedSolution(ctx)
+	if err != nil {
+		return err
+	}
+
+	sc.resources[beforeDiffField] = initiallyAppliedSolution
+
+	return nil
+}
+
+func (sc *SaptuneChangeSolution) commit(ctx context.Context) error {
+	initiallyAppliedSolution, _ := sc.resources[beforeDiffField].(string)
+
+	if initiallyAppliedSolution == "" {
+		return fmt.Errorf(
+			"cannot change solution to %s because no solution is currently applied",
+			sc.parsedArguments.solution,
+		)
+	}
+
+	if sc.parsedArguments.solution == initiallyAppliedSolution {
+		sc.logger.Infof("solution %s is already applied. Nothing to change, skipping commit phase", sc.parsedArguments.solution)
+		return nil
+	}
+
+	return sc.saptune.ChangeSolution(ctx, sc.parsedArguments.solution)
+}
+
+func (sc *SaptuneChangeSolution) verify(ctx context.Context) error {
+	initiallyAppliedSolution, _ := sc.resources[beforeDiffField].(string)
+
+	if sc.parsedArguments.solution == initiallyAppliedSolution {
+		sc.resources[afterDiffField] = initiallyAppliedSolution
+		return nil
+	}
+
+	appliedSolution, err := sc.saptune.GetAppliedSolution(ctx)
+	if err != nil {
+		return err
+	}
+
+	if appliedSolution != sc.parsedArguments.solution {
+		return fmt.Errorf(
+			"verify saptune apply failing, the solution %s was not applied in commit phase",
+			sc.parsedArguments.solution,
+		)
+	}
+	sc.resources[afterDiffField] = appliedSolution
+	return nil
+}
+
+func (sc *SaptuneChangeSolution) rollback(ctx context.Context) error {
+	initiallyAppliedSolution, _ := sc.resources[beforeDiffField].(string)
+
+	if initiallyAppliedSolution == "" {
+		return nil
+	}
+
+	sc.logger.Infof("Reverting the requested solution: %s", sc.parsedArguments.solution)
+	if err := sc.saptune.RevertSolution(ctx, sc.parsedArguments.solution); err != nil {
```

Comment:
I don't think we need to do the `revert`.
The `change` will already revert the already existing solution (no matter if the one that is not wrong or the previous one). It should be smart enough.
So, i think just doing the next `change` should be enough